src/bleanser/modules/iter.py

```diff
@@ -0,0 +1,58 @@
+from contextlib import contextmanager
+from pathlib import Path
+from typing import Iterator, Any
+
+
+from bleanser.core.processor import BaseNormaliser
+
+
+class IterNormaliser(BaseNormaliser):
+    """
+    This is meant to be overridden by a subclass
+
+    parse_file receives an input file, and should yield data/objects that when converted
+    to a string, produces some comparable data/object to the cleaned output file
+
+    possible things this could return is a unique key/id, or a tuple of (key, data), or a
+    namedtuple/dataclass
+
+    newlines are stripped from the string, so lines can be compared/diffed properly
+
+    Its possible you could use a library or code from https://github.com/karlicoss/HPI
+    in parse_file, to use the DAL itself to parse the file https://beepb00p.xyz/exports.html#dal
+    """
+
+    def parse_file(self, path: Path) -> Iterator[Any]:
+        raise NotImplementedError
+        # when you subclass, you should do something like
+        # with path.open('r') as f:
+        #   for object in some_library(f):
+        #       yield (object.id, object.key)
+
+    def _emit_history(self, upath: Path, cleaned) -> None:
+        """
+        calls parse_file to extract lines from the unpacked path
+        subclasses should override that to yield some kind of object
+        out to here
+        """
+        with cleaned.open("w") as f:
+            for line in self.parse_file(upath):
+                # newlines may interfere with the diffing, as we sort the lines
```

Comment:
another option could be something like `f.write(repr(str(line)))`? it should make sure there is no weird stuff that ends up on the terminal